### PR TITLE
Don't login when clicking on app button on login screen

### LINKED ISSUE #2424: Link Buttons on Login page trigger login
To replicate:

(Occurs in Chrome and Firefox)
1. Type in your login credentials into the Login screen
2. Press one of the three buttons at the bottom (Android, Apple, F-Droid)
3. The Login will be triggered without ever pressing the actual Login Button.

Expected behavior: The Buttons shouldn't trigger a login...

### PR DESCRIPTION
**Pull request by johnbotris**
fix #2424

The app buttons were triggering form submit, call preventDefault to stop this. Also switched them over to use ButtonN since i was there

### CODE REVIEW

File: src/gui/base/ButtonN.js
Comment (charlag) [inline src/gui/base/ButtonN.js:107]: You already get event in `click`, just do `preventDefault` there

File: src/login/LoginView.js
Comment (charlag) [inline src/login/LoginView.js:45]: We don't need to save them into array, just render them inline, maybe in a separate method

**Review by charlag — CHANGES_REQUESTED** [review]

File: src/gui/base/ButtonN.js
Comment (johnbotris) [reply src/gui/base/ButtonN.js:107]: can do, just thought it helped to make the ButtonAttrs instances look more streamlined

**Review by charlag — APPROVED** [review]

**Review by jowlo — COMMENTED** [review]
looks all good

**Comment by jowlo** [comment]
Sorry for the mess, for some reason i cannot approve this PR. Maybe because it is already merged anyway.

**Comment by charlag** [comment]
It's merged and you self-requested the review?

**Comment by jowlo** [comment]
Yeah, that was a bit stupid. I thought I needed to self-request the review to be able to approve. I looked over the code and tested so I thought I might as well approve in addition to the `tested` tag, as I... well... approved.

But apparently you cannot approve once it is merged.

**Pull request merged**
